src/shogun/mathematics/linalg/backend/eigen/BasicOps.cpp

```diff
@@ -266,6 +291,25 @@ void LinalgBackendEigen::pinv_impl(
 	result_eig = V_eig * s_eig.asDiagonal() * U_eig.transpose();
 }
 
+template <typename T>
+void LinalgBackendEigen::rank_update_impl(
+    SGMatrix<T>& A, const SGVector<T>& b, T alpha) const
+{
+	T x;
+	T update;
+	for (auto j : range(A.num_cols))
+	{
+		x = b[j];
```

Comment:
Out of curiosity, having the local x is better than just using `b[j]` all over the method? Same question for `update`.